server/chat-plugins/rock-paper-scissors.tsx

```diff
@@ -44,11 +47,14 @@ export class RPSGame extends Rooms.RoomGame {
 		this.players = [];
 
 		this.room.update();
-		this.room.send(`|controlshtml|<center>Waiting for another player to join....</center>`);
+		this.controls(<div style={{textAlign: 'center'}}>Waiting for another player to join....</div>)
 		this.sendField();
 	}
+	controls(node: preact.VNode) {
+		this.room.send(`|controlshtml|` + Chat.renderNode(node));
+	}
 	onConnect(user: User, connection: Connection) {
-		this.room.sendUser(connection, this.getField());
+		this.room.sendUser(connection, '|fieldhtml|' + Chat.renderNode(this.getField()));
```

Comment:
I think maybe `Chat.renderNode` should be built into `Chat.html`.

So you could just do

``Chat.html`|fieldhtml|${this.getField()}` ``

You'd need some way of detecting Preact components, though... I could see it being difficult.

Reply:
Nah, it's not that hard. Done.